src/tests/ftest/nvme/enospace.py

```diff
@@ -0,0 +1,448 @@
+        # read in loop, until storage space is full.
+        out_queue = queue.Queue()
+        job = threading.Thread(target=self.ior_bg_thread,
+                               kwargs={"results": out_queue})
+        job.daemon = True
+        job.start()
+
+        #Run IOR in Foreground
+        self.run_enospace_foreground()
+        # Verify the background job queue and make sure no FAIL for any IOR run
+        while not self.out_queue.empty():
+            if self.out_queue.get() == "FAIL":
+                self.fail("One of the Background IOR job failed")
+
+    def test_enospace_lazy_with_bg(self):
+        """Jira ID: DAOS-4756.
+
+        Test Description: IO gets DER_NOSPACE when SCM and NVMe is full with
+                          default (lazy) Aggregation mode.
+
+        Use Case: This tests will create the pool and fill 75% of SCM size which
+                  will trigger the aggregation because of space pressure,
+                  next fill 75% more which should fill NVMe. Try to fill 60%
+                  more and now SCM size will be full too.
+                  verify that last IO fails with DER_NOSPACE and SCM/NVMe pool
+                  capacity is full.One background IO job will be running
+                  continuously.
+
+        :avocado: tags=all,hw,medium,nvme,ib2,full_regression
+        :avocado: tags=der_enospace,enospc_lazy,enospc_lazy_bg
+        """
+        print(self.pool.pool_percentage_used())
+
+        #Run IOR to fill the pool.
+        self.run_enospace_with_bg_job()
+
+    def test_enospace_lazy_with_fg(self):
+        """Jira ID: DAOS-4756.
+
+        Test Description: Fill up the system (default aggregation mode) and
+                          delete all containers in loop, which should release
+                          the space.
+
+        Use Case: This tests will create the pool and fill 75% of SCM size which
+                  will trigger the aggregation because of space pressure,
+                  next fill 75% more which should fill NVMe. Try to fill 60%
+                  more and now SCM size will be full too.
+                  verify that last IO fails with DER_NOSPACE and SCM/NVMe pool
+                  capacity is full. Delete all the containers.
+                  Do this in loop for 10 times and verify space is released.
+
+        :avocado: tags=all,hw,medium,nvme,ib2,full_regression
+        :avocado: tags=der_enospace,enospc_lazy,enospc_lazy_fg
+        """
+        print(self.pool.pool_percentage_used())
+
+        #Repeat the test in loop.
+        for _loop in range(10):
+            print("-------enospc_lazy_fg Loop--------- {}".format(_loop))
+            #Run IOR to fill the pool.
+            self.run_enospace_foreground()
+            #Delete all the containers
+            self.delete_all_containers()
+            #Delete container will take some time to release the space
+            time.sleep(60)

```

Comment:
Hmm, Okay. I was just thinking if we are going to sleep it will be quite a lot of time being consumed every time. 10*60 secs, whereas it reality way less time might only be required in total for space return. It's your call. Just putting it out there.